src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -512,6 +511,26 @@ abstract class ManagedStrategy(
         onThreadFinish(currentThreadId)
     }
 
+    /**
+     * Handles exceptions that occur in a specific thread.
+     * This method is called when a thread finishes with an exception.
+     *
+     * @param exception The exception that was thrown within the thread.
+     * @return true if the exception should be suppressed, false otherwise.
+     */
+    override fun onThreadException(exception: Throwable): Boolean = runInIgnoredSection {
+        val currentThreadId = threadScheduler.getCurrentThreadId()
```

Comment:
Please add `check(isInternalException(..)) { .. }` to highlight the contract

Reply:
Here I use `if (isInternalException(...))` to delegate to function `onInternalException` (which is also used in actors case). Otherwise the exception is just normal exception of `Thread::run`, and should not be suppressed, so I `return false` in this case.